NOnion/Directory/ServerDescriptorsDocument.fs

```diff
@@ -1,4 +1,7 @@
-﻿namespace NOnion.Directory
+﻿// Types in this file are marked as obsolete and this file has self-references which cause pointless warnings
```

Comment:
@aarani I don't understand, why dont you follow your same advice that you give in the Obsolete attribute?

Reply:
what? This file uses the types inside itself (there are multiple types that use each other) to be able to parse server descriptor files.

Reply:
Server descriptor classes are used in the test project, we can't just mark some of them as internal they are records using each other.

Reply:
the tests that use Obsolete stuff, shouldn't we remove them? otherwise it's not obsolete

Reply:
> the tests that use Obsolete stuff, shouldn't we remove them? otherwise it's not obsolete

It's not a test, it's a helper function that finds random routers for a couple of tests(CircuitHelper.cs), I can use TorDirectory to replace it but it would make more tests dependent on TorDirectory.